django/contrib/contenttypes/management.py

```diff
@@ -1,9 +1,56 @@
 from django.apps import apps
-from django.db import DEFAULT_DB_ALIAS, router
+from django.db import DEFAULT_DB_ALIAS, migrations, router, transaction
+from django.db.utils import IntegrityError
 from django.utils import six
 from django.utils.six.moves import input
 
 
+class RenameContentType(migrations.RunPython):
+    def __init__(self, app_label, old_name, new_name):
+        self.app_label = app_label
+        self.old_name = old_name
+        self.new_name = new_name
+        super(RenameContentType, self).__init__(self.rename_forward, self.rename_backward)
+
+    def _rename(self, apps, old_name, new_name):
+        ContentType = apps.get_model('contenttypes', 'ContentType')
+        try:
+            content_type = ContentType.objects.get_by_natural_key(self.app_label, old_name.lower())
+        except ContentType.DoesNotExist:
+            pass
+        else:
+            content_type.model = new_name.lower()
+            using = router.db_for_write(ContentType, instance=content_type)
+            try:
+                with transaction.atomic(using=using):
+                    content_type.save(update_fields={'model'})
+            except IntegrityError:
```

Comment:
Alright, do you think a `RunTimeWarning` or `logger.exception` would also be a good idea?